### ISSUE TITLE
When interacting with a ExpansionTile using a screen reader, the status of the tile is not announced

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#120650]
Internal bug: b/269139224

(Copied from the internal bug above)

The bug also happens on iOS devices.

## Steps to Reproduce

1. With an Android device or emulator with Talkback enabled.
2. Focus on the collapsed/expanded ExpansionTile and verify what the screen reader announces.
3. Double click to activate/collapse the ExpansionTile and verify the screen reader announcement.

**Expected results:** 

The tile should read "double tap to expand", "double tap to collapse" when focused on it.
The tile should announce "expanded" or "collapsed" when clicking on it.

**Actual results:** 

Screen reader is just reading "double tap to activate" when focused on the tile.
The tile doesn't announce anything when it's expanded or collapsed.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        // We also apply a custom font, ProductSans.
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: ExpansionTile(
          title: Text('Tile'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

**Comment by TahaTesser** [comment]
Filed https://github.com/flutter/flutter/pull/120881 to fix this.

**Comment by TahaTesser** [comment]
https://github.com/flutter/flutter/pull/120881 is reverted in https://github.com/flutter/flutter/pull/121624

**Comment by TahaTesser** [comment]
cc: @chunhtai 
Assigning this to you as well.

(In my investigation, it looks like this needs a complete fix, including regular `ListTile` and `ExpansionTile` don't announce "double tap to" action https://github.com/flutter/flutter/issues/121636).
